documentation/src/main/asciidoc/userguide/appendices/LegacyBasicTypeResolution.adoc

```diff
@@ -0,0 +1,346 @@
+----
+====
+
+Alternatively, you can use the `@TypeDef` and skip the registration phase:
+
+[[basic-custom-type-BitSetTypeDef-mapping-example]]
+.Using `@TypeDef` to register a custom Type
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/basic/BitSetTypeDefTest.java[tags=basic-custom-type-BitSetTypeDef-mapping-example]
+----
+====
+
+To validate this new `BasicType` implementation, we can test it as follows:
+
+[[basic-custom-type-BitSetType-persistence-example]]
+.Persisting the custom `BasicType`
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/basic/BitSetTypeTest.java[tags=basic-custom-type-BitSetType-persistence-example]
+----
+====
+
+When executing this unit test, Hibernate generates the following SQL statements:
+
+[[basic-custom-type-BitSetType-persistence-sql-example]]
+.Persisting the custom `BasicType`
+====
+[source, JAVA, indent=0]
+----
+include::{originalextrasdir}/basic/basic-custom-type-BitSetType-persistence-sql-example.sql[]
+----
+====
+
+As you can see, the `BitSetType` takes care of the _Java-to-SQL_ and _SQL-to-Java_ type conversion.
+
+[[basic-custom-type-UserType]]
+==== Implementing a `UserType`
+
+The second approach is to implement the `UserType` interface.
+
+[[basic-custom-type-BitSetUserType-example]]
+.Custom `UserType` implementation
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/basic/BitSetUserType.java[tags=basic-custom-type-BitSetUserType-example]
+----
+====
+
+The entity mapping looks as follows:
+
+[[basic-custom-type-BitSetUserType-mapping-example]]
+.Custom `UserType` mapping
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/basic/BitSetUserTypeTest.java[tags=basic-custom-type-BitSetUserType-mapping-example]
+----
+====
+
+In this example, the `UserType` is registered under the `bitset` name, and this is done like this:
+
+[[basic-custom-type-register-UserType-example]]
+.Register a Custom `UserType` implementation
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/basic/BitSetUserTypeTest.java[tags=basic-custom-type-register-UserType-example]
+----
+
+or using the `MetadataBuilder`
+
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/../bootstrap/BootstrapTest.java[tags=basic-custom-type-register-UserType-example]
+----
+====
+
+[NOTE]
+====
+Like `BasicType`, you can also register the `UserType` using a simple name.
+
+Without registering a name, the `UserType` mapping requires the fully qualified class name:
+
+[source, JAVA, indent=0]
+----
+@Type( type = "org.hibernate.userguide.mapping.basic.BitSetUserType" )
+----
+====
+
+When running the previous test case against the `BitSetUserType` entity mapping, Hibernate executed the following SQL statements:
+
+[[basic-custom-type-BitSetUserType-persistence-sql-example]]
+.Persisting the custom `BasicType`
+====
+[source, JAVA, indent=0]
+----
+include::{originalextrasdir}/basic/basic-custom-type-BitSetUserType-persistence-sql-example.sql[]
+----
+====
```

Comment:
this file should be added to the list of included chapter into the `Hibernate_User_Guide.adoc`